From dc58026a621fa892afdbba8de81e6c246b080664 Mon Sep 17 00:00:00 2001
From: Mike Beaton <mjsbeaton@gmail.com>
Date: Sat, 8 Nov 2025 06:41:18 +0000
Subject: [PATCH 33/33] SecurityPkg: Remove unused variable

XCODE5 toolchain finds this, with error:
Tpm2Capability.c:535:23: error: variable 'ActivePcrBankCount' set but not used

Fixes: https://github.com/tianocore/edk2/commit/1f2adcbba571faead4795bfbeeb232e82166c901

Signed-off-by: Mike Beaton <mjsbeaton@gmail.com>
---
 SecurityPkg/Library/Tpm2CommandLib/Tpm2Capability.c | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/SecurityPkg/Library/Tpm2CommandLib/Tpm2Capability.c b/SecurityPkg/Library/Tpm2CommandLib/Tpm2Capability.c
index b15f72cabaa6..5dcac49d5ff9 100644
--- a/SecurityPkg/Library/Tpm2CommandLib/Tpm2Capability.c
+++ b/SecurityPkg/Library/Tpm2CommandLib/Tpm2Capability.c
@@ -532,13 +532,11 @@ Tpm2GetCapabilitySupportedAndActivePcrs (
   EFI_STATUS          Status;
   TPML_PCR_SELECTION  Pcrs;
   UINTN               Index;
-  UINT8               ActivePcrBankCount;
 
   //
   // Get supported PCR
   //
-  Status             = Tpm2GetCapabilityPcrs (&Pcrs);
-  ActivePcrBankCount = 0;
+  Status = Tpm2GetCapabilityPcrs (&Pcrs);
   //
   // If error, assume that we have at least SHA-1 (and return the error.)
   //
@@ -546,7 +544,6 @@ Tpm2GetCapabilitySupportedAndActivePcrs (
     DEBUG ((DEBUG_ERROR, "GetSupportedAndActivePcrs - Tpm2GetCapabilityPcrs fail!\n"));
     *TpmHashAlgorithmBitmap = HASH_ALG_SHA1;
     *ActivePcrBanks         = HASH_ALG_SHA1;
-    ActivePcrBankCount      = 1;
   }
   //
   // Otherwise, process the return data to determine what algorithms are supported
@@ -563,7 +560,6 @@ Tpm2GetCapabilitySupportedAndActivePcrs (
           if (!IsZeroBuffer (Pcrs.pcrSelections[Index].pcrSelect, Pcrs.pcrSelections[Index].sizeofSelect)) {
             DEBUG ((DEBUG_VERBOSE, "GetSupportedAndActivePcrs - HASH_ALG_SHA1 active.\n"));
             *ActivePcrBanks |= HASH_ALG_SHA1;
-            ActivePcrBankCount++;
           }
 
           break;
@@ -573,7 +569,6 @@ Tpm2GetCapabilitySupportedAndActivePcrs (
           if (!IsZeroBuffer (Pcrs.pcrSelections[Index].pcrSelect, Pcrs.pcrSelections[Index].sizeofSelect)) {
             DEBUG ((DEBUG_VERBOSE, "GetSupportedAndActivePcrs - HASH_ALG_SHA256 active.\n"));
             *ActivePcrBanks |= HASH_ALG_SHA256;
-            ActivePcrBankCount++;
           }
 
           break;
@@ -583,7 +578,6 @@ Tpm2GetCapabilitySupportedAndActivePcrs (
           if (!IsZeroBuffer (Pcrs.pcrSelections[Index].pcrSelect, Pcrs.pcrSelections[Index].sizeofSelect)) {
             DEBUG ((DEBUG_VERBOSE, "GetSupportedAndActivePcrs - HASH_ALG_SHA384 active.\n"));
             *ActivePcrBanks |= HASH_ALG_SHA384;
-            ActivePcrBankCount++;
           }
 
           break;
@@ -593,7 +587,6 @@ Tpm2GetCapabilitySupportedAndActivePcrs (
           if (!IsZeroBuffer (Pcrs.pcrSelections[Index].pcrSelect, Pcrs.pcrSelections[Index].sizeofSelect)) {
             DEBUG ((DEBUG_VERBOSE, "GetSupportedAndActivePcrs - HASH_ALG_SHA512 active.\n"));
             *ActivePcrBanks |= HASH_ALG_SHA512;
-            ActivePcrBankCount++;
           }
 
           break;
@@ -603,7 +596,6 @@ Tpm2GetCapabilitySupportedAndActivePcrs (
           if (!IsZeroBuffer (Pcrs.pcrSelections[Index].pcrSelect, Pcrs.pcrSelections[Index].sizeofSelect)) {
             DEBUG ((DEBUG_VERBOSE, "GetSupportedAndActivePcrs - HASH_ALG_SM3_256 active.\n"));
             *ActivePcrBanks |= HASH_ALG_SM3_256;
-            ActivePcrBankCount++;
           }
 
           break;
-- 
2.52.0

